tests/store/artifact/test_unity_catalog_models_artifact_repo.py

```diff
@@ -0,0 +1,208 @@
+)
+
+
+# TODO: remove this mock once the UC model registry store is supported
+@pytest.fixture(autouse=True, scope="module")
+def mock_get_databricks_unity_catalog_store():
+    def get_uc_rest_store(store_uri):
+        return UcModelRegistryStore(store_uri)
+
+    with mock.patch(
+        "mlflow.tracking._model_registry.utils._get_databricks_uc_rest_store"
+    ) as _get_databricks_uc_rest_store_mock:
+        _get_databricks_uc_rest_store_mock.side_effect = get_uc_rest_store
+        yield _get_databricks_uc_rest_store_mock
+
+
+def test_uc_models_artifact_repo_init_with_uri_containing_profile():
+    uri_with_profile = "models://profile@databricks-uc/MyModel/12"
+    models_repo = UnityCatalogModelsArtifactRepository(
+        uri_with_profile, _DATABRICKS_UNITY_CATALOG_SCHEME
+    )
+    assert models_repo.artifact_uri == uri_with_profile
+    assert models_repo.client._registry_uri == f"{_DATABRICKS_UNITY_CATALOG_SCHEME}://profile"
+
+
+def test_uc_models_artifact_repo_init_with_db_profile_inferred_from_context():
+    uri_without_profile = "models:/MyModel/12"
+    profile_in_registry_uri = "some_profile"
+    registry_uri = f"databricks-uc://{profile_in_registry_uri}"
+    models_repo = UnityCatalogModelsArtifactRepository(
+        artifact_uri=uri_without_profile, registry_uri=registry_uri
+    )
+    assert models_repo.artifact_uri == uri_without_profile
+    assert models_repo.client._registry_uri == registry_uri
+
+
+def test_uc_models_artifact_repo_init_not_using_databricks_registry_raises():
+    non_databricks_uri = "non_databricks_uri"
+    model_uri = "models:/MyModel/12"
+    with pytest.raises(
+        MlflowException,
+        match="Attempted to instantiate an artifact repo to access models in the Unity "
+        "Catalog with non-Unity Catalog registry URI",
+    ):
+        UnityCatalogModelsArtifactRepository(model_uri, non_databricks_uri)
+
+
+def test_uc_models_artifact_repo_with_stage_uri_raises():
+    model_uri = "models:/MyModel/Staging"
+    with pytest.raises(
+        MlflowException, match="staged-based model URIs are unsupported for models in UC"
+    ):
+        UnityCatalogModelsArtifactRepository(model_uri, _DATABRICKS_UNITY_CATALOG_SCHEME)
+
+
+def test_uc_models_artifact_uri_with_scope_and_prefix_throws():
+    with pytest.raises(
+        MlflowException,
+        match="Remote model registry access via model URIs of the form "
+        "'models://<scope>@<prefix>/<model_name>/<version_or_stage>'",
+    ):
+        UnityCatalogModelsArtifactRepository(
+            "models://scope:prefix@databricks-uc/MyModel/12", _DATABRICKS_UNITY_CATALOG_SCHEME
+        )
+
+
+def _mock_temporary_creds_response(temporary_creds):
+    mock_response = mock.MagicMock(autospec=Response)
+    mock_response.status_code = 200
+    mock_response.text = json.dumps({"credentials": temporary_creds})
+    return mock_response
+
+
+def test_uc_models_artifact_repo_download_artifacts_uses_temporary_creds_aws():
```

Comment:
I was thinking about parametrizing this across clouds via `pytest.mark.parametrize` but it seemed difficult; we need to mock different artifact repository classes (e.g. `S3ArtifactRepository`, `AzureDataLakeArtifactRepository`), and inspect their constructor args